android-core/src/main/kotlin/com/mparticle/internal/messages/BaseMPMessage.kt

```diff
@@ -0,0 +1,74 @@
+package com.mparticle.internal.messages
+
+import android.location.Location
+import com.mparticle.internal.Constants
+import com.mparticle.internal.InternalSession
+import com.mparticle.internal.MPUtility
+import org.json.JSONException
+import org.json.JSONObject
+
+open class BaseMPMessage : JSONObject {
+    var mpId: Long = 0
+        private set
+
+    protected constructor()
+
+    constructor(builder: BaseMPMessageBuilder, session: InternalSession, location: Location?, mpId: Long) : super(builder, builder.keys) {
```

Comment:
I see in the original file this was explicitly marked `protected`, is that not necessary?

Reply:
Kotlin: A protected constructor is not accessible to other classes in the same package, only in subclasses. and we are using in same package class BaseMPMessageBuilder.